azurerm/resource_arm_container_group.go

```diff
@@ -0,0 +1,294 @@
+package azurerm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/arm/containerinstance"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceArmContainerGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmContainerGroupCreate,
+		Read:   resourceArmContainerGroupRead,
+		Update: resourceArmContainerGroupCreate,
+		Delete: resourceArmContainerGroupDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"ip_address_type": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
```

Comment:
I'm assuming there's a limited set of options here? if so - could we add some validation to this, e.g.
```
ValidateFunc: validation.StringInSlice([]string{
  "foo",
  "bar",
}, true)
```

Reply:
(as below) we should be able to make this `Optional` and `Default: "Public"` given `Public" is the only option at this time, which would mean users don't need to specify it :)

Reply:
Done